binary-search-tree/ruby/inventory_management.rb

```diff
@@ -34,17 +34,31 @@ def initialize
 
   def search(id)
     # Comment out the line below and make changes here
-    raise "you must implement search"
+    # raise "you must implement search"
+    node = @root
+
+    if node == nil
+      nil
+    elsif node.val == id
+      node
+    elsif id < node.val
+      node = node.left
+    else 
+      node = node.right
+    end
```

Comment:
Great work so far with your implementation of the `search` method! It's clear that you understand the idea of checking each node and making a decision to accept it, move left or move right based on the search value.

However, what's missing is the iterative or recursive aspect of the search. To complete your implementation, you need to continue the search process until you either find the target node or reach the end of the tree.

To achieve this, you can choose to use a loop or a recursive function. If you choose to use a loop, you can set the current node to the root of the tree and continue moving left or right until you either find the target node or reach a nil node. If you choose to use recursion, you can define a recursive function that takes the current node as a parameter, checks if it is nil, checks if its value is equal to the target value, and then recursively calls itself with the left or right child of the current node depending on the value of the target.

I hope this helps clarify what you need to do next. Keep up the good work!